cl/recap/mergers.py

```diff
@@ -1387,8 +1391,7 @@ def merge_pacer_docket_into_cl_docket(
     return rds_created, content_updated
 
 
-@transaction.atomic
```

Comment:
I'm a pretty huge fan of this transaction. Why remove it?

Reply:
transaction isn't really async orm compatible, we have this constraint on `RECAPDocument` so it didn't really look like it was actually needed:
```python
        unique_together = (
            "docket_entry",
            "document_number",
            "attachment_number",
        )
```

Reply:
Boy, I don't know. I'm not happy about it, but I think if we can't put this in a transaction, it's not worth making it async. There are all kinds of opportunities for this method to fail halfway through. I really like knowing at this point that if that happens at least the DB won't be messed up. If it's a choice between correctness and performance, I have to choose correctness.

Reply:
> I'm not happy about it, but I think if we can't put this in a transaction, it's not worth making it async. 

Keep in mind this could be a significant hot path when we add support for processing attachments from dockets.

> There are all kinds of opportunities for this method to fail halfway through.

I'm not really seeing any that are problematic, am I missing something here?

> I really like knowing at this point that if that happens at least the DB won't be messed up. If it's a choice between correctness and performance, I have to choose correctness.

I'm just not seeing what the reason for a transaction is in this case since we have a unique constraint on the attachment rows and using `aupdate_or_create` should be safe without a transaction AFAIU due to the constraint.

Reply:
> I'm not really seeing any that are problematic, am I missing something here?

Yeah, OK. I read over it carefully too, and I guess I just really don't like it from a good practices perspective. You're right though, I can't see anywhere that'd be that terrible if it crashed. We could have some incomplete merging, or we could have an item pushed to S3 that we don't wind up merging, but all of that is more or less OK. We don't wind up with corrupt data at least. 

Someday, somebody is going to tweak this in a way where a transaction would be necessary, and they're not going to have one, and I'm going to forget to insist on one, and we'll have corrupt data, so that's a bummer, but I guess for now it's fine.

Reply:
> I just really don't like it from a good practices perspective

AFAIU avoiding transactions when they are unnecessary is kind of a good practice from a performance optimization perspective.

Reply:
Yeah, that's true. I think I'm in the "When in doubt..." camp.